### ISSUE TITLE
GCP integrations fail when adding multiple projects

### ISSUE THREAD

**Issue by jake-desco** [DataDog/terraform-provider-datadog#1369]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version

Terraform v1.0.11
on linux_amd64
+ provider registry.terraform.io/datadog/datadog v3.8.1

### Affected Resource(s)

 datadog_integration_gcp

### Terraform Configuration Files

```hcl
source "datadog_integration_gcp" "gcp_integration" {
    for_each = var.gcp_projects

    project_id     = each.key
    private_key_id = local.gcp_acct_info.private_key_id
    private_key    = local.gcp_acct_info.private_key
    client_email   = local.gcp_acct_info.client_email
    client_id      = local.gcp_acct_info.client_id

    host_filters   = "datadog:yes"
}

```

### Debug Output

The debug output contains sensitive data, I can't paste all of it.  But here is an redacted request in the output that is failing:

```
POST /api/v1/integration/gcp HTTP/1.1
Host: api.datadoghq.com
User-Agent: terraform-provider-datadog/3.8.1 (terraform 2.10.0; terraform-cli 1.0.11) datadog-api-client-go/1.8.0 (go go1.16.13; os linux; arch amd64)
Content-Length: 2330
Accept: application/json
Content-Type: application/json
Dd-Api-Key: REDACTED
Dd-Application-Key: REDACTED
Dd-Operation-Id: CreateGCPIntegration
Accept-Encoding: gzip

{"auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","auth_uri":"https://accounts.google.com/o/oauth2/auth","automute":false,"c
lient_email":"REDACTED","client_id":"115255351018288812947","client_x509_cert_url":"https://www.googleap
is.com/robot/v1/metadata/x509/REDACTED","host_filters":"datadog:yes","private_key":"-----BEGIN PRIVATE K
EY-----REDACTED\n-----END PRIVATE KEY-----\n","private_key_id":"REDACTED","project_id":"REDACTED","token_uri":"https://oauth2.googleapis.com/token","type":"service_account"}: timestamp=2022-02-24T15:05:14.855-0500
2022-02-24T15:05:14.971-0500 [INFO]  provider.terraform-provider-datadog_v3.8.1: 2022/02/24 15:05:14
HTTP/1.1 409 Conflict
Transfer-Encoding: chunked
Cache-Control: no-cache
Connection: keep-alive
Content-Security-Policy: frame-ancestors 'self'; report-uri REDACTED
Content-Type: application/json
Date: Thu, 24 Feb 2022 20:05:14 GMT
Pragma: no-cache
Strict-Transport-Security: max-age=15724800;
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Ratelimit-Limit: 12000
X-Ratelimit-Period: 60
X-Ratelimit-Remaining: 11949
X-Ratelimit-Reset: 46

4c
{"errors": ["This Service account already exists in your Datadog account."]}
0
```

### Expected Behavior

The projects should all be added.

### Actual Behavior

Each time the apply is run a small number of the projects manage to get added.  The rest return the 409 Conflict error.

### Steps to Reproduce

1. Create a service account in GCP in a central project
2. Grant IAM roles as per https://docs.datadoghq.com/integrations/google_cloud_platform/ step 4, at the GCP Org level (so that it has access to all of the projects)
3. Use that service account to add all projects using a datadog_integration_gc resource in a loop using foreach

**Comment by skarimo** [comment]
This should be resolved in the next release.  Fix was introduces in the following PR https://github.com/DataDog/terraform-provider-datadog/pull/1360

**Comment by leseguillonmael** [comment]
hi @skarimo, 
When will be the next release ?  thx

**Comment by jake-desco** [comment]
> This should be resolved in the next release. Fix was introduces in the following PR #1360

This is great, thank you.  Interesting fix - I was trying to achieve the same thing with parallelism=1 but failed.  Out of interest though, this looks like it is really an issue with the DataDog APIs not allowing concurrent requests, so fixing in TF is a bandaid really -- will it be fixed in the backend?

**Comment by skarimo** [comment]
@leseguillonmael should be within the next week or so.

@jake-desco there are internal discussions to better handle concurrent requests. It would also be helpful if you could open a ticket with datadog support so it is tracked by the appropriate team. Datadog support: https://www.datadoghq.com/support
